packages/compass-sidebar/src/components/sidebar-databases-navigation.ts

```diff
@@ -28,7 +29,7 @@ function mapStateToProps(state: any) {
     ])
   );
   const isReadOnly =
-    process.env.HADRON_READONLY === 'true' ||
+    preferencesReadOnly ||
```

Comment:
Had a good conversation with @gribnoysup about these cases where we keep track of the preference value in the stores in this PR, and how we could avoid having to do that – not blocking for this PR, but more in the “it might be nice not to have to add tracking to the stores” kind of way:

In most cases, accessing preferences falls into two categories:
1. Affecting how something is displayed in the UI
2. Affecting how an action is performed in response to a UI event

In the first case, the `usePreference` hook should get us fairly far, in the second case, it should be fine to access the current value of the preference in the place in which it is actually used via `getPreferences().<name>`.

Taking this as an example, I think it would be pretty okay to drop the readonly checks from this package entirely, and instead do a check inside the target component itself, something along the lines of:

```diff
diff --git a/packages/compass-databases-navigation/src/databases-navigation-tree.tsx b/packages/compass-databases-navigation/src/databases-navigation-tree.tsx
index edd5a27a6311..44cad773beaa 100644
--- a/packages/compass-databases-navigation/src/databases-navigation-tree.tsx
+++ b/packages/compass-databases-navigation/src/databases-navigation-tree.tsx
@@ -84,12 +84,13 @@ const NavigationItem = memo<{
   // eslint-disable-next-line @typescript-eslint/unbound-method
   const {
     items,
-    isReadOnly,
     activeNamespace,
     currentTabbable,
     onDatabaseExpand,
     onNamespaceAction,
   } = data;
+  const preferencesReadOnly = usePreference('readOnly');
+  const isReadOnly = preferencesReadOnly || data.isReadOnly;
 
   const itemData = items[index];
 ```